sycl/doc/extensions/supported/sycl_ext_intel_buffer_location.asciidoc

```diff
@@ -45,7 +45,7 @@ The use of this extension requires a target that supports SPV_INTEL_fpga_buffer_
 == Overview
 
 On targets that provide more than one type of global memory, knowing that a particular pointer can only access one of those memory types at compile time can enable compiler optimizations.  
-This extension adds an accessor property to indicate to the runtime in which of these memories the buffer corresponding to this accessor should be allocated and to inform the compiler that all accesses made through that accessor can only ever interact with the given memory.
+This extension adds an accessor property to indicate to the runtime in which of these memories the buffer corresponding to this accessor should be allocated and to inform the compiler that all accesses made through that accessor can only ever interact with the given memory or it can be used to inform the usm alocation (`malloc_device`, `malloc_shared`, `malloc_host`) to allocate on a given memory.
 This information is not a hint; it is a functional requirement of the program that must be respected.
 
 == Modifications to the SYCL 2020 Pre-Provisional Specification
```

Comment:
The description of the property (below) is currently very specific to accessors:

>  The buffer_location property notifies the SYCL device compiler that the given accessor will only ever point to the memory identified by the int template parameter of its instance class. It also notifies the SYCL runtime to store the given accessor in that memory.

Can you update this so it makes sense also for USM allocations?

Reply:
Thanks for the suggestion! The specifics of how this is going to interact with the malloc api depend on the which workaround we choose.

Given that we are only choosing between 1 and 3, malloc api will be updated to align with https://github.com/intel/llvm/pull/5656

Reply:
Thanks! I have added some more examples, and description for passing this property into usm allocation APIs.